homeassistant/components/tuya/binary_sensor.py

```diff
@@ -348,6 +353,48 @@ async def async_setup_entry(
     """Set up Tuya binary sensor dynamically through Tuya discovery."""
     hass_data = entry.runtime_data
 
+    def _get_fault_labels(device: CustomerDevice) -> list[str]:
+        """Return fault labels for the device."""
+        if DPCode.FAULT not in device.status_range:
+            return []
+
+        fault_type = DPType(device.status_range[DPCode.FAULT].type)
+        if fault_type != DPType.BITMAP:
+            return []
+
+        fault_values = json.loads(device.status_range[DPCode.FAULT].values)
+        if "label" not in fault_values:
+            return []
```

Comment:
These can be combined.

Reply:
I've refactored the code a bit after moving to `homeassistant.util.json` as the types differed a bit.
Please resolve this comment if you think it looks all right now.